src/services/MethodDependencyBuilder.ts

```diff
@@ -0,0 +1,132 @@
+import MethodParser from "./MethodParser";
+import GithubService from "./GithubService";
+import {Method} from "../interfaces/Method";
+import GitCommitCache from "./GitCommitCache";
+import RestClient from "../rest/RestClient";
+import {CommitInfo, FileInfo} from "../interfaces/GitHubTypes";
+
+/**
+ * A class to build cross-cut dependencies (across previous commits) between methods of HEAD.
+ */
+export default class MethodDependencyBuilder {
+    methodParser: MethodParser;
+    ghService: GithubService;
+    liveRestClient: RestClient;
+    liveCache: GitCommitCache;
+
+    constructor() {
+        this.methodParser = new MethodParser();
+        this.liveRestClient = new RestClient();
+        this.liveCache = new GitCommitCache();
+        this.ghService = new GithubService(this.liveRestClient, this.liveCache);
```

Comment:
I guess the only think here is the fact that we have multiple commit cache instances around - I don't think it'll be too much of a problem, but we'll go to disk often.

Reply:
Yeah we should really clean up / re-evaluate how we wanna use ghService and cache but out of scope for this PR so we can get some features going

Reply:
Yeah 🚢it